scene/2d/animated_sprite_2d.cpp

```diff
@@ -476,6 +476,8 @@ void AnimatedSprite2D::play(const StringName &p_name, float p_custom_scale, bool
 	int end_frame = MAX(0, frames->get_frame_count(animation) - 1);
```

Comment:
The fix is correct as is, but I think it makes sense to move this line to the `else` block below (so `MAX(0, frames->get_frame_count(animation) - 1)` would not be done twice when `name != animation`).

(same for 3d)

Reply:
Yes but you would still need one for the else branch, but you're right in that it would not be called twice

Reply:
That's what was suggested, to move the one that's outside the `if` to the `else`

Reply:
Yeah, I'm suggesting to have separate variables within both `if`/`else` blocks (with single assignment each).